TestControllers/AzureController.psm1

```diff
@@ -48,6 +48,9 @@ Class AzureController : TestController
 		$this.TestProvider.PlatformFaultDomainCount = $this.CustomParams["PlatformFaultDomainCount"]
 		$this.TestProvider.PlatformUpdateDomainCount = $this.CustomParams["PlatformUpdateDomainCount"]
 		$this.TestProvider.EnableTelemetry = $ParamTable["EnableTelemetry"]
+		if ($this.CustomParams["EnableNSG"] -and $this.CustomParams["EnableNSG"] -eq "true") {
+			$this.TestProvider.EnableNSG = $true
+		}
```

Comment:
Just use customparams can enable NSG?

Reply:
then AzureProvider and Azure.psm have logic to add nsg rules.